pkg/apis/garden/v1beta1/conversions.go

```diff
@@ -95,13 +95,15 @@ func Convert_garden_MachineImage_To_v1beta1_MachineImage(in *garden.MachineImage
 func Convert_v1beta1_KubernetesConstraints_To_garden_KubernetesConstraints(in *KubernetesConstraints, out *garden.KubernetesConstraints, s conversion.Scope) error {
 	out.OfferedVersions = []garden.KubernetesVersion{}
 	duplicates := map[string]int{}
-	for index, externalVersion := range in.Versions {
+	index := 0
```

Comment:
Yes. The loop index variable is used currently, which causes a shift of the expiration date if there are duplicated Kubernetes versions. The index should only be increased when a version is appended to `out.OfferedVersions` because it is stored in the `duplicates` map and used in [Line 117](https://github.com/gardener/gardener/pull/1568/files/5aa35248d39d95a68def9490c717f83131e4b9cf#diff-75a7ffe069df78188e5ab6c32c26a995R117) to set the corresponding expirationDate. The current behaviour causes an index out of range error, when a duplicated version is specified and the last version has an expiration date set.